adapters/blis/blis.go

```diff
@@ -0,0 +1,135 @@
+	"github.com/prebid/prebid-server/v3/errortypes"
+	"github.com/prebid/prebid-server/v3/macros"
+	"github.com/prebid/prebid-server/v3/openrtb_ext"
+	"github.com/prebid/prebid-server/v3/util/iterutil"
+	"github.com/prebid/prebid-server/v3/util/jsonutil"
+)
+
+type adapter struct {
+	endpointTemplate *template.Template
+}
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	endpointTemplate, err := template.New("endpointTemplate").Parse(config.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse endpoint url template: %v", err)
+	}
+
+	bidder := &adapter{
+		endpointTemplate: endpointTemplate,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var impExt adapters.ExtImpBidder
+	if err := jsonutil.Unmarshal(request.Imp[0].Ext, &impExt); err != nil {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("Invalid imp.ext for impression index %d. Error Infomation: %s", 0, err.Error()),
+		}}
+	}
+	var impExtBidder openrtb_ext.ImpExtBlis
+	if err := jsonutil.Unmarshal(impExt.Bidder, &impExtBidder); err != nil {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("Invalid imp.ext.bidder for impression index %d. Error Infomation: %v", 0, err),
+		}}
+	}
+
+	endpoint, err := a.buildEndpointURL(&impExtBidder)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	headers := http.Header{}
+	headers.Add("X-Supply-Partner-Id", impExtBidder.SupplyPartnerID)
+
+	requestJSON, err := jsonutil.Marshal(request)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	requestData := &adapters.RequestData{
+		Method:  "POST",
+		Uri:     endpoint,
+		Headers: headers,
+		Body:    requestJSON,
+		ImpIDs:  openrtb_ext.GetImpIDs(request.Imp),
+	}
+
+	return []*adapters.RequestData{requestData}, nil
+}
+
+func (a *adapter) buildEndpointURL(impExtBidder *openrtb_ext.ImpExtBlis) (string, error) {
+	endpointParams := macros.EndpointTemplateParams{SupplyId: impExtBidder.SupplyPartnerID}
+	return macros.ResolveMacros(a.endpointTemplate, endpointParams)
+}
+
+func (a *adapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.RequestData, responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if adapters.IsResponseStatusCodeNoContent(responseData) {
+		return nil, nil
+	}
+
+	if err := adapters.CheckResponseStatusCodeForErrors(responseData); err != nil {
+		return nil, []error{err}
+	}
+
+	var response openrtb2.BidResponse
+	if err := jsonutil.Unmarshal(responseData.Body, &response); err != nil {
+		return nil, []error{err}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(request.Imp))
+	bidResponse.Currency = response.Cur
+	var errors []error
+	for seatBid := range iterators.SlicePointerValues(response.SeatBid) {
+		for bid := range iterators.SlicePointerValues(seatBid.Bid) {
```

Comment:
Now that #4447 merged, can you update this branch and you probably need to change these
```suggestion
	for seatBid := range iterutil.SlicePointerValues(response.SeatBid) {
		for bid := range iterutil.SlicePointerValues(seatBid.Bid) {
```